services/backend/src/assignments-controller.js

```diff
@@ -80,7 +80,14 @@ async function deleteAssignment(req, res) {
     await dynamoUtils.delete({
         TableName: dynamoUtils.TABLES.ASSIGNMENTS,
         Key: { aid: requestedAid },
-        ExpressionAttributeValues: { ':uid': req.auth.uid },
+        ExpressionAttributeValues: { ':uid': req.params.uid },
+        ConditionExpression: 'uid = :uid',
+    });
+
+    await dynamoUtils.delete({
+        TableName: dynamoUtils.TABLES.TASKS,
+        Key: { aid: requestedAid },
+        ExpressionAttributeValues: { ':uid': req.params.uid },
```

Comment:
A delete operation in Dynamo only deletes a single item by key. Check out [batchWrite](https://docs.aws.amazon.com/AWSJavaScriptSDK/latest/AWS/DynamoDB/DocumentClient.html#batchWrite-property) for deleting multiple items.

This is honestly kinda complicated because you gotta handle pagination, I think it'd warrant its own pull request